text/0055-remove-unspecified.md

```diff
@@ -0,0 +1,248 @@
+
+### Naming specifics
+
+The names `RequestEntity` and `__cedar::Default` are both subject to change.
+
+For the latter, we have agreed that we want to use the `__cedar` prefix (reserved as part of [cedar#557](https://github.com/cedar-policy/cedar/pull/557) in v3.1), but the exact type name is up for debate.
+Options we've considered include:
+
+- `None`
+- `Null`
+- `Ghost`
+- `Dummy`
+- `Empty`
+- `Irrelevant`
+- `Generic`
+
+Here are some options we’ve ruled out:
+
+- `Any`: possibility to confuse the scope condition `principal` with `principal is __cedar::Any`. The first applies to any principal, while the latter applies to only the default entity.
+- `Arbitrary`: same issue as "any"
+- `Unspecified`: potential for confusion with partial evaluation "unknown"
+- `Mock`: implies that this entity type should be used for debugging/testing
+
+### Upgrade strategy
+
+This RFC proposes a **breaking change** for both the API and the language.
+It breaks the API because it requires modifying the `Request` constructor, and it breaks the language because it changes the schema format.
+Fortunately, this proposal allows for a straightforward period of deprecation.
+
+To start, we will leave the current APIs and schema format as-is, while also adding the new `__cedar::Default` type and support for using it in schemas.
+Omitting an `appliesTo` field in the schema will become equivalent to having `[__cedar::Default]`, but we will emit a warning to recommend updating to the new syntax.
+
+Then, in a subsequent major version, we will implement the breaking changes described above.
+
+### Documentation
+
+As pointed out in reviews of this RFC, a key reason why unspecified/default entities are confusing is that we have almost no documentation for them.
+Thus, an important part of this RFC is developing useful documentation.
+We expect this to include a blog post with examples of applications where you may want to use a default entity, and recommendations for best practices.
+
+## Drawbacks
+
+Breaking changes are painful for users.
+We need to make sure that we provide enough value in the major version with this change for the upgrade to be worth it.
+
+## Alternatives
+
+### Alternative A - Get rid of the unspecified entity type
+
+The current proposal can be viewed as a re-framing of the current unspecified type.
+A more radical option (as originally proposed by this RFC) is to completely drop Cedar support for unspecified entities, instead recommending that users create their own application-specific entity types that act like unspecified entities.
+
+This is (arguably) the approach we used in the [TinyTodo application](https://github.com/cedar-policy/cedar-examples/tree/release/3.0.x/tinytodo): we made a special `Application::“TinyTodo”` entity to act as the resource for the “CreateList” and “GetLists” actions.
```

Comment:
This RFC specifically forbids mentioning the default entity in policies, but TinyTodo policies do mention `Application::"TinyTodo"` explicitly:
```
permit(principal, action == Action::"CreateList", resource == Application::"TinyTodo");
```
If we think these things are the same, then we should allow the default entity to be mentioned in policies. Per my comment above, I think that's a modification in the wrong direction: We should be even _more_ restrictive on the use of default entities, not less.